npm/pkg/dataplane/ipsets/ipsetmanager.go

```diff
@@ -208,6 +232,46 @@ func (iMgr *IPSetManager) ApplyIPSets(networkID string) error {
 	return nil
 }
 
+func (iMgr *IPSetManager) GetIPsFromSelectorIPSets(setList map[string]struct{}) (map[string]struct{}, error) {
+	if len(setList) == 0 {
+		return map[string]struct{}{}, nil
+	}
+	iMgr.Lock()
+	defer iMgr.Unlock()
+
+	setintersections := make(map[string]struct{})
+	var err error
+	firstLoop := true
+	for setName := range setList {
+		if !iMgr.exists(setName) {
+			return nil, fmt.Errorf("[ipset manager] selector ipset %s does not exist", setName)
+		}
+		set := iMgr.setMap[setName]
+		if firstLoop {
+			intialSetIPs := set.IPPodKey
+			for k := range intialSetIPs {
+				setintersections[k] = struct{}{}
+			}
+			firstLoop = false
+		}
+		setintersections, err = set.getSetIntersection(setintersections)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return setintersections, err
+}
+
+func (iMgr *IPSetManager) GetNetworkPolicyReferencesBySet(setName string) (map[string]struct{}, error) {
+	iMgr.Lock()
+	defer iMgr.Unlock()
+	if !iMgr.exists(setName) {
+		return nil, fmt.Errorf("[ipset manager] selector ipset %s does not exist.", setName)
+	}
+	set := iMgr.setMap[setName]
+	return set.SelectorReference, nil
```

Comment:
return set.NetPolReference?

Reply:
Sorry i corrected this, we need to return selectorreferences here, i corrected all variables in the caller function.